### PR TITLE
TRUNK-5848: Remove raw use of parameterized collections

### PR DESCRIPTION
**Pull request by sacull**
## Description of what I changed

My goal was to find collections that were not parameterized. After that, I removed the cast, which became redundant after parameterizing the collection. I also improved the style of the edited files according to https://wiki.openmrs.org/display/docs/Java+Conventions in some files by replacing huge amounts of spaces with tabs. Ctrl+Alt+L is your friend. ;)

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5848

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/notification/impl/MessageServiceImpl.java [hunk 1]
```diff
@@ -258,7 +258,7 @@ public Message prepareMessage(Template template) throws MessageException {
 	@Transactional(readOnly = true)
 	public Message prepareMessage(String templateName, Map data) throws MessageException {
 		try {
-			Template template = (Template) getTemplatesByName(templateName).get(0);
+			Template template = getTemplatesByName(templateName).get(0);
```
Comment (HerbertYiga) [inline api/src/main/java/org/openmrs/notification/impl/MessageServiceImpl.java:261]: Was curios to see how you ended up with this line,good i saw it in the above class.

File: api/src/test/java/liquibase/ext/change/core/InsertWithUuidDataChangeDatabaseIT.java [hunk 2]
```diff
@@ -50,8 +50,8 @@ public void shouldInsertUuids() throws Exception {
 		}
 	}
 	
-	protected Map getNamesWithUuids() throws SQLException {
-		Map<String, String> result = new HashMap();
+	protected Map<String, String> getNamesWithUuids() throws SQLException {
```
Comment (HerbertYiga) [inline api/src/test/java/liquibase/ext/change/core/InsertWithUuidDataChangeDatabaseIT.java:53]: LGTM! cc @dkayiwa   @ibacher

File: api/src/main/java/org/openmrs/PatientProgram.java [hunk 3]
```diff
@@ -51,7 +51,7 @@
 	
 	private Set<PatientState> states = new HashSet<>();
          
-        private Set<PatientProgramAttribute> attributes = new LinkedHashSet();
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/PatientProgram.java:54]: Did you accidentally format (TAB) the above?

Comment (sacull) [reply api/src/main/java/org/openmrs/PatientProgram.java:54]: @dkayiwa You're right, I didn't notice it. It definitely needs to be corrected.

File: api/src/main/java/org/openmrs/notification/MessageService.java [hunk 4]
```diff
@@ -82,9 +82,9 @@ public Message createMessage(String recipients, String sender, String subject, S
 	public Message prepareMessage(Template template) throws MessageException;
 	
 	// Template methods
-	public List getAllTemplates() throws MessageException;
+	public List<Template> getAllTemplates() throws MessageException;
```
[the remark this answers is not in the document]
Comment (sacull) [reply api/src/main/java/org/openmrs/notification/MessageService.java:85]: @dkayiwa I don't understand that. Interfaces only have public methods, so it's probably okay.

[the remark this answers is not in the document]
Comment (dkayiwa) [reply api/src/main/java/org/openmrs/notification/MessageService.java:85]: https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Deprecation

[the remark this answers is not in the document]
Comment (sacull) [reply api/src/main/java/org/openmrs/notification/MessageService.java]: @dkayiwa Forgive me for the delay, but I couldn't figure out how to deal with this problem for a long time. On the one hand, the returned list should be parameterized, on the other hand, I can't just change the return type in the interface as you wrote above. Nor can I mark it as deprecated as I would have to add a new one that would need to be implemented throughout the project. At the present stage of my knowledge of the OpenMRS code, I prefer not to make such deep changes. All in all, at the moment it is better if this piece of code remains the old one.

**Comment by dkayiwa** [comment]
Fair enough. So are you just going to close the pull request?

**Comment by sacull** [comment]
@dkayiwa I've undone this one change, but GitHub shows it's still there. Well, there are only problems with this pull request. I guess it really needs to be closed.

By the way, based on SonarQube, I posted a few more code improvements on the issues pages. I wouldn't be angry if there were only a few minutes to allow them to be implemented. ;)

**Pull request closed without merging**